addons/test_mail/static/tests/activity_tests.js

```diff
@@ -231,62 +214,62 @@ QUnit.test('activity view: activity widget', async function (assert) {
     const mailActivityTypeIds = pyEnv['mail.activity.type'].search([]);
     const [mailTestActivityId2] = pyEnv['mail.test.activity'].search([['name', '=', 'Office planning']]);
     const [mailTemplateId1] = pyEnv['mail.template'].search([['name', '=', 'Template1']]);
-    const actionServiceInterceptor = ev => {
-            var action = ev.data.action;
+    const actionServiceInterceptor = makeActionServiceInterceptor({
+        doAction(action, options, originalDoAction) {
             if (action.serverGeneratedAction) {
                 assert.step('serverGeneratedAction');
             } else if (action.res_model === 'mail.compose.message') {
                 assert.deepEqual({
                     default_model: 'mail.test.activity',
-                    default_res_id: mailTestActivityId2,
+                    default_res_id: mailActivityTypeIds[1],
                     default_template_id: mailTemplateId1,
                     default_use_template: true,
                     force_email: true
                     }, action.context);
                 assert.step("do_action_compose");
             } else if (action.res_model === 'mail.activity') {
                 assert.deepEqual({
-                    "default_activity_type_id": mailActivityTypeIds[1],
+                    "default_activity_type_id": mailTestActivityId2,
                     "default_res_id": mailTestActivityId2,
                     "default_res_model": 'mail.test.activity',
                 }, action.context);
                 assert.step("do_action_activity");
-            } else {
+            } else if (action.res_model !== 'mail.test.activity') {
```

Comment:
extra condition ?
else unreachable?

Reply:
This one is reachable, the else before checks `res_model === 'mail.activity'`, this ones checks that `res_model !== 'mail.test.activity`. It's done this way because the `openView` action is expected.

Reply:
Ok for that one, but the return after the else should be definitely unreachable